apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/RareSats/useRareSatsModel.tsx

```diff
@@ -0,0 +1,53 @@
+import { mockedRareSats } from "LLD/features/Collectibles/__integration__/mockedRareSats";
+import { mappingKeysWithIconAndName } from "../Icons";
+import { MappingKeys } from "LLD/features/Collectibles/types/Ordinals";
+import { IconProps } from "LLD/features/Collectibles/types/Collection";
+import { RareSat } from "LLD/features/Collectibles/types/Ordinals";
+
+type RareSatsProps = {};
+
+/** THIS HOOK WILL BASICLY NOT BE THE SAME AT ALL ONCE WE PLUG THE API*/
+export const useRareSatsModel = (_props: RareSatsProps) => {
+  const processedRareSats: RareSat[] = mockedRareSats.map(sat => {
```

Comment:
`processedRareSats` can be more easier to read if you split into small function for example a solution could be something like that


```

const getKeyFromDisplayName = (displayName: string): MappingKeys => 
  displayName.toLowerCase().replace(" ", "_") as MappingKeys;

const generateNameAndIcons = (displayName: string | string[]): { name: string, icons: { [key: string]: ({ size, color, style }: IconProps) => JSX.Element } } => {
  const icons: { [key: string]: ({ size, color, style }: IconProps) => JSX.Element } = {};
  let name: string;

  if (Array.isArray(displayName)) {
    name = displayName
      .map(dn => mappingKeysWithIconAndName[getKeyFromDisplayName(dn)]?.name)
      .join(" / ");
    displayName.forEach(dn => {
      const key = getKeyFromDisplayName(dn);
      icons[key] = mappingKeysWithIconAndName[key]?.icon;
    });
  } else {

    const key = getKeyFromDisplayName(displayName);
    const mapping = mappingKeysWithIconAndName[key];

    icons[key] = mapping?.icon;
    name = mapping?.name;
  }

  return { name, icons };
};

const processedRareSats: RareSat[] = mockedRareSats.map(sat => {
  const { display_name } = sat;
  const { name, icons } = generateNameAndIcons(display_name);

  return {
    ...sat,
    icons,
    name,
  };
});

```

Reply:
It wasn't the correct hook I've changed it